src/tsf/ConsoleTSF.cpp

```diff
@@ -52,17 +52,17 @@ HRESULT CConsoleTSF::Initialize()
     Init_CheckResult();
 
     // Set the context owner before attaching the context to the doc.
-    CComQIPtr<ITfSource> spSrcIC(_spITfInputContext);
+    wil::com_ptr_nothrow<ITfSource> spSrcIC(_spITfInputContext.try_query<ITfSource>());

```

Comment:
Note that the original code is a bit weird here. It queries for `ITfSource` but it does not check if the query succeeded. At least not before using it on the next line to call `AdviseSink`.

However, it does check `spSrcIC` on line 83 below. Obviously, it would never reach that line if the interface isn't present.

Reply:
Changed to
```
    wil::com_ptr_nothrow<ITfSource> spSrcIC;
    hr = _spITfInputContext.query_to(&spSrcIC);
    Init_CheckResult();
```